modules/RTC/BridgeChannel.js

```diff
@@ -0,0 +1,320 @@
+        // event.
+        if (peerconnection) {
+            peerconnection.ondatachannel = event => {
+                // NOTE: We assume that the "onDataChannel" event just fires
+                // once.
+
+                const datachannel = event.channel;
+
+                // Handle the RTCDataChannel.
+                this._handleChannel(datachannel);
+                this._mode = 'datachannel';
+            };
+
+        // Otherwise create a WebSocket connection.
+        } else if (wsUrl) {
+            // Create a WebSocket instance.
+            const ws = new WebSocket(wsUrl);
+
+            // Handle the WebSocket.
+            this._handleChannel(ws);
+            this._mode = 'websocket';
+        }
+    }
+
+    /**
+     * The channel mode.
+     * @return {string} "datachannel" or "websocket" (or null if not yet set).
+     */
+    get mode() {
+        return this._mode;
+    }
+
+    /**
+     * Closes the currently opened channel.
+     */
+    close() {
+        if (this._channel) {
+            try {
+                this._channel.close();
+            } catch (error) {} // eslint-disable-line no-empty
+
+            this._channel = null;
+        }
+    }
+
+    /**
+     * Whether there is an underlying RTCDataChannel or WebSocket and it's
+     * open.
+     * @return {boolean}
+     */
+    isOpen() {
+        return this._channel && this._channel.readyState === 'open';
+    }
+
+    /**
+     * Sends message via the channel.
+     * @param {string} to The id of the endpoint that should receive the
+     * message. If "" the message will be sent to all participants.
+     * @param  {object} payload The payload of the message.
+     * @throws NetworkError or InvalidStateError from RTCDataChannel#send (@see
+     * {@link https://developer.mozilla.org/docs/Web/API/RTCDataChannel/send})
+     * or from WebSocket#send or Error with "No opened channel" message.
+     */
+    sendMessage(to, payload) {
+        this._send({
+            colibriClass: 'EndpointMessage',
+            msgPayload: payload,
+            to
+        });
+    }
+
+    /**
+     * Sends a "lastN value changed" message via the channel.
+     * @param {number} value The new value for lastN. -1 means unlimited.
+     */
+    sendSetLastNMessage(value) {
+        const jsonObject = {
+            colibriClass: 'LastNChangedEvent',
+            lastN: value
+        };
+
+        this._send(jsonObject);
+        logger.log(`Channel lastN set to: ${value}`);
+    }
+
+    /**
+     * Sends a "pinned endpoint changed" message via the channel.
+     * @param {string} endpointId The id of the pinned endpoint.
+     * @throws NetworkError or InvalidStateError from RTCDataChannel#send (@see
+     * {@link https://developer.mozilla.org/docs/Web/API/RTCDataChannel/send})
+     * or from WebSocket#send or Error with "No opened channel" message.
+     */
+    sendPinnedEndpointMessage(endpointId) {
+        this._onXXXEndpointChanged('pinned', endpointId);
```

Comment:
while we are here, refactoring, can you please inline that function here? it's unnecessarily complicated.